### PR TITLE
git clone check branch exists and support slashes

### PR DESCRIPTION
**Pull request by kobymeir**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6693

## Description
git clone check branch exists and support slashes

### CODE REVIEW

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 1]
```diff
@@ -196,14 +198,28 @@
...
 
-
 .trigger-slack-notification:
   stage: .post
   trigger:
     strategy: depend
     include:
       - local: .gitlab/ci/.gitlab-ci.slack-notify.yml
 
+.validate_content_test_conf_merged:
```
Comment (tkatzir) [inline .gitlab/ci/.gitlab-ci.global.yml:208]: We care about a specific branch.
```suggestion
.validate_content_test_conf_branch_merged:
```

File: Tests/scripts/download_conf_repos.sh [hunk 2]
```diff
@@ -23,18 +32,41 @@ clone_repository() {
...
+    user_info="${user}:${token}@"
+  fi
+  git ls-remote --exit-code --quiet --heads "https://${user_info}${host}/${repo_name}.git" "refs/heads/${branch}" 1>/dev/null 2>&1
+  local branch_exists=$?
+
+  if [ "${branch_exists}" -ne 0 ]; then
+    echo "Branch ${branch} does not exist in ${repo_name}, defaulting to ${fallback_branch}"
+    local exit_code=1
+  else
+    echo "Branch ${branch} exists in ${repo_name} trying to clone!"
```
Comment (tkatzir) [inline Tests/scripts/download_conf_repos.sh:59]:
```suggestion
    echo "Branch ${branch} exists in ${repo_name}, trying to clone"
```

File: Tests/scripts/download_conf_repos.sh [hunk 3]
```diff
@@ -23,18 +32,41 @@ clone_repository() {
...
+    clone_repository "${host}" "${user}" "${token}" "${repo_name}" "${branch}" "${retry_count}" "${sleep_time}"
+    local exit_code=$?
+    if [ "${exit_code}" -ne 0 ]; then
+      echo "Failed to clone ${repo_name} with branch:${branch}, exit code:${exit_code}"
+    fi
+  fi
+  if [ "${exit_code}" -ne 0 ]; then
+    # Trying to clone from fallback branch.
+    echo "Trying to clone repository:${repo_name} with branch ${fallback_branch}!"
```
Comment (tkatzir) [inline Tests/scripts/download_conf_repos.sh:68]:
```suggestion
    echo "Trying to clone repository:${repo_name} with fallback branch ${fallback_branch}!"
```

File: Tests/scripts/download_conf_repos.sh [hunk 4]
```diff
@@ -49,10 +81,7 @@ clone_repository_with_fallback_branch() {
   fi
 }
 
-# Replace slashes '/' in the branch name with underscores '_'.
-UNDERSCORE_BRANCH=${CI_COMMIT_BRANCH//\//_}
-
-echo "Getting conf from branch ${UNDERSCORE_BRANCH} (With fallback to master)"
+echo "Getting content-test-conf and infra repositories with branch:${CI_COMMIT_BRANCH}, With fallback to master"
```
Comment (tkatzir) [inline Tests/scripts/download_conf_repos.sh:84]:
```suggestion
echo "Getting content-test-conf and infra repositories with branch:${CI_COMMIT_BRANCH}, with fallback to master"
```

**Review by tkatzir — APPROVED** [review]
Great!
Approved, with comments.

File: Tests/scripts/download_conf_repos.sh [hunk 5]
```diff
@@ -69,7 +98,7 @@ echo ${DEMISTO_LIC_PATH} > demisto_lic_path
 DEMISTO_PACK_SIGNATURE_UTIL_PATH="./signDirectory"
 echo ${DEMISTO_PACK_SIGNATURE_UTIL_PATH} > demisto_pack_sig_util_path
 
-clone_repository_with_fallback_branch "content-test-conf" "${UNDERSCORE_BRANCH}" "master" 3
+clone_repository_with_fallback_branch "code.pan.run" "gitlab-ci-token" "${CI_JOB_TOKEN}" "xsoar/content-test-conf" "${CI_COMMIT_BRANCH}" 3 10 "master"
```
[the remark this answers is not in the document]
Comment (kobymeir) [reply Tests/scripts/download_conf_repos.sh:101]: yes, sadly, that's a hard coded value for gitlab. 
https://docs.gitlab.com/ee/ci/jobs/ci_job_token.html

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 6]
```diff
@@ -333,26 +349,7 @@
...
-            RED='\033[0;31m'
-            NC='\033[0m'
-            echo -e "${RED}ERROR: Found a branch with the same name in contest-test-conf conf.json - $UNDERSCORE_CI_BRANCH.\n Merge it in order to merge the current branch into content repo.${NC}"
-            job-done
-            exit 1
-          fi
-        }
-      fi
-    - section_end "Validate content-test-conf Branch Merged"
+    - !reference [ .validate_content_test_conf_merged ]
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.global.yml:352]:
```suggestion
    - !reference [ .validate_content_test_conf_branch_merged ]
```

File: .gitlab/ci/.gitlab-ci.bucket-upload.yml [hunk 7]
```diff
@@ -133,11 +133,13 @@ xpanse-prepare-testing-bucket-upload-flow:
...
-    - section_start "Download configuration"
-    - ./Tests/scripts/download_conf_repos.sh  >> $ARTIFACTS_FOLDER/logs/installations.log
+    - section_start "Download content-test-conf and infra" --collapsed
+    - ./Tests/scripts/download_conf_repos.sh 2>&1 | tee --append "${ARTIFACTS_FOLDER}/logs/download_conf_repos.log"
+    - section_end "Download content-test-conf and infra"
+    - section_start "Fetching secrets" --collapsed
     - SECRET_CONF_PATH=$(cat secret_conf_path)
     - python3 ./Tests/scripts/add_secrets_file_to_build.py -sa "$GSM_SERVICE_ACCOUNT" -sf "$SECRET_CONF_PATH" -u "$DEMISTO_USERNAME" -p "$DEMISTO_PASSWORD" -gpidd "$GSM_PROJECT_ID_DEV" -gpidp "$GSM_PROJECT_ID" >> $ARTIFACTS_FOLDER/logs/handle_secrets.log
-    - section_end "Download configuration"
+    - section_end "Fetching secrets"
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.bucket-upload.yml:142]:
```suggestion
    - section_end "Secrets Fetch"
```

File: .gitlab/ci/.gitlab-ci.bucket-upload.yml [hunk 8]
```diff
@@ -207,11 +209,13 @@ install-packs-in-server-master:
...
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.bucket-upload.yml:218]:
```suggestion
    - section_end "Secrets Fetch"
```

File: .gitlab/ci/.gitlab-ci.bucket-upload.yml [hunk 9]
```diff
@@ -207,11 +209,13 @@ install-packs-in-server-master:
     - .default-job-settings
   script:
     - EXIT_CODE=0
-    - section_start "Download configuration"
-    - ./Tests/scripts/download_conf_repos.sh  >> $ARTIFACTS_FOLDER/logs/installations.log
+    - section_start "Download content-test-conf and infra" --collapsed
+    - ./Tests/scripts/download_conf_repos.sh 2>&1 | tee --append "${ARTIFACTS_FOLDER}/logs/download_conf_repos.log"
+    - section_end "Download content-test-conf and infra"
+    - section_start "Fetching secrets" --collapsed
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.bucket-upload.yml:215]:
```suggestion
    - section_start "Secrets Fetch" --collapsed
```

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 10]
```diff
@@ -53,11 +53,13 @@
   - section_end "Create ID Set"
 
 .download-demisto-conf:
-  - section_start "Download content-test-conf" --collapsed
-  - ./Tests/scripts/download_conf_repos.sh  >> $ARTIFACTS_FOLDER/logs/download_demisto_conf.log
+  - section_start "Download content-test-conf and infra" --collapsed
+  - ./Tests/scripts/download_conf_repos.sh 2>&1 | tee --append "${ARTIFACTS_FOLDER}/logs/download_conf_repos.log"
+  - section_end "Download content-test-conf and infra"
+  - section_start "Fetching secrets" --collapsed
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.global.yml:59]:
```suggestion
  - section_start "Secrets Fetch" --collapsed
```

File: .gitlab/ci/.gitlab-ci.global.yml [hunk 11]
```diff
@@ -53,11 +53,13 @@
...
   - SECRET_CONF_PATH=$(cat secret_conf_path)
   - python3 ./Tests/scripts/add_secrets_file_to_build.py -sa "$GSM_SERVICE_ACCOUNT" -sf "$SECRET_CONF_PATH" -u "$DEMISTO_USERNAME" -p "$DEMISTO_PASSWORD" -gpidd "$GSM_PROJECT_ID_DEV" -gpidp "$GSM_PROJECT_ID" >> $ARTIFACTS_FOLDER/logs/handle_secrets.log
-  - section_end "Download content-test-conf"
+  - section_end "Fetching secrets"
```
Comment (kobymeir) [inline .gitlab/ci/.gitlab-ci.global.yml:62]:
```suggestion
  - section_end "Secrets Fetch"
```

**Pull request merged**